### PR TITLE
feat(*): add upgrade command and validations

### PR DESCRIPTION
**Pull request by michelleN**

### CODE REVIEW

File: cmd/tiller/release_server.go [hunk 1]
```diff
@@ -174,7 +174,54 @@ func (s *releaseServer) GetReleaseContent(c ctx.Context, req *services.GetReleas
 }
 
 func (s *releaseServer) UpdateRelease(c ctx.Context, req *services.UpdateReleaseRequest) (*services.UpdateReleaseResponse, error) {
-	return nil, errNotImplemented
```
Comment (adamreese) [inline cmd/tiller/release_server.go]: `errNotImplemented` is no longer used, you can delete the definition

File: cmd/helm/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		chart: ch,
+	})
+
+	// update chart version
+	cfile = &chart.Metadata{
+		Name:        "testUpgradeChart",
+		Description: "A Helm chart for Kubernetes",
+		Version:     "0.1.2",
+	}
+	chartPath, err = chartutil.Create(cfile, tmpChart)
```
Comment (adamreese) [inline cmd/helm/upgrade_test.go]: err is not used

File: cmd/tiller/release_server.go [hunk 3]
```diff
@@ -174,7 +174,54 @@ func (s *releaseServer) GetReleaseContent(c ctx.Context, req *services.GetReleas
...
+	givenChart := req.Chart.Metadata.Name
+	releasedChart := rel.Chart.Metadata.Name
+	if givenChart != releasedChart {
+		return nil, fmt.Errorf("Given chart, %s, does not match chart originally released, %s", givenChart, releasedChart)
+	}
+
+	// validate new chart version is higher than old
+	givenChartVersion := req.Chart.Metadata.Version
+	releasedChartVersion := rel.Chart.Metadata.Version
+	if givenChartVersion <= releasedChartVersion {
```
Comment (adamreese) [inline cmd/tiller/release_server.go]: Using comparison operators on strings compares lexically byte-wise.  Check out https://github.com/Masterminds/semver#checking-version-constraints

File: cmd/helm/upgrade.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		return err
+	}
+
+	_, err = u.client.UpdateRelease(u.release, chartPath, helm.UpdateValueOverrides(rawVals), helm.UpgradeDryRun(u.dryRun))
+	if err != nil {
+		return prettyError(err)
+	}
+
+	fmt.Fprintf(u.out, "It's not you. It's me\nYour upgrade looks valid but this command is still under active development.\nHang tight.\n")
```
Comment (adamreese) [inline cmd/helm/upgrade.go]: Can I at least get my CD's back?

**Comment by adamreese** [comment]
minor ocd remove the TODO at `// UpdateOption -- TODO`

File: cmd/helm/upgrade.go (hunk 4 above)
Comment (michelleN) [inline cmd/helm/upgrade.go]: na :)

File: cmd/helm/helm_test.go [hunk 5]
```diff
@@ -108,7 +131,7 @@ func (c *fakeReleaseClient) ReleaseStatus(rlsName string, opts ...helm.StatusOpt
 	return nil, nil
 }
 
-func (c *fakeReleaseClient) UpdateRelease(rlsName string, opts ...helm.UpdateOption) (*rls.UpdateReleaseResponse, error) {
+func (c *fakeReleaseClient) UpdateRelease(rlsName string, chStr string, opts ...helm.UpdateOption) (*rls.UpdateReleaseResponse, error) {
```
Comment (adamreese) [inline cmd/helm/helm_test.go]: what does `chStr` represent?

Comment (michelleN) [inline cmd/helm/helm_test.go]: path of a chart you want to upgrade to

**Pull request merged**